agent/global.go

```diff
@@ -28,6 +28,10 @@ const (
 	// heartbeat to API server.
 	defaultHeartbeatInterval = 30 * time.Second
 
+	// defaultHeartbeatTimeout is how long the agent can perform operations when
```

Comment:
I intentionally made this short because the default 6 hour exec timeout is too generous a timeout. This is only meant to fill in the gaps where the task is in between commands, so there's no explicit user-configurable timeout that it can use. If it's using this default, tasks have 8 minutes (= 1 minute + 7 minutes grace period) to do anything, which is fairly generous, given that the task shouldn't be kept waiting for more than a couple seconds between commands.